src/messaging.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * collection of functions to show messages with actions in a consistent manner
+ */
+
+import * as vscode from 'vscode'
+
+export interface MessageAction {
+  title: string
+  action: () => void
+}
+
+export function systemErrorMessage(message: string, ...actions: Array<MessageAction>) {
+  const msg = vscode.window.showErrorMessage(message, ..._extractActionTitles(actions))
+  if (msg) {
```

Comment:
In which case would `showErrorMessage` return something falsy? According to the documentation it returns `Thenable<string | undefined>` which, before being resolved, is always truthy.